desktop/preload.js

```diff
@@ -1,26 +1,32 @@
 const { contextBridge, ipcRenderer } = require('electron');
 
+const validChannels = [
+  'appCommand',
+  'clearCookies',
+  'importNotes',
+  'notesImported',
+  'setAutoHideMenuBar',
+  'settingsUpdate',
+  'wpLogin',
+];
+
 contextBridge.exposeInMainWorld('electron', {
   send: (channel, data) => {
     // whitelist channels
-    let validChannels = [
-      'appCommand',
-      'clearCookies',
-      'setAutoHideMenuBar',
-      'settingsUpdate',
-      'wpLogin',
-    ];
     if (validChannels.includes(channel)) {
       ipcRenderer.send(channel, data);
     }
   },
-  receive: (channel, func) => {
-    let validChannels = ['appCommand', 'wpLogin'];
+  receive: (channel, callback) => {
+    if (validChannels.includes(channel)) {
+      const newCallback = (_, data) => callback(data);
+      ipcRenderer.on(channel, newCallback);
+    }
+  },
+  once: (channel, callback) => {
     if (validChannels.includes(channel)) {
-      // Deliberately strip event as it includes `sender`
-      ipcRenderer.on(channel, (event, ...args) => {
-        return func(...args);
-      });
+      const newCallback = (_, data) => callback(data);
```

Comment:
why strip out the event value? it seems odd to intentionally hide it.

in any case, looking at the type signature I see `callback` taking two arguments, the `event` and `notes` but this `newCallback` specifically drops anything past the second argument, so it's almost like we're preventing the app from behaving the way we're encoding it. I'm sure I might be missing something though.

if the issue were simply that we didn't want to need to worry about `event` in `addNotesToApp` I'd definitely be more in favor of managing that specific detail when passing `addNotesToApp` in as the callback so that we don't cause its particular need to ripple into other cases of `once`

```ts
window.electron?.once('notesImported', (event, notes) => addNotesToApp(notes));
```

Reply:
This is intentional to not expose the event to the non-electron side of things. The event includes the `sender`. Here are the details: https://www.electronjs.org/docs/api/ipc-main#sending-messages